vcd/resource_vcd_vapp_vm.go

```diff
@@ -926,16 +970,37 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}) er
 
 	}
 
-	if d.HasChanges("memory", "cpus", "cpu_cores", "power_on", "disk", "expose_hardware_virtualization",
-		"network", "boot_image", "hardware_version", "os_type", "description") {
+	memoryNeedToChange := true
+	cpusNeedToChange := true
+	if executionType == "update" {
+		if d.Get("memory_hot_add_enabled").(bool) && d.HasChange("memory") {
+			memoryNeedToChange = false
+		} else if !d.HasChange("memory") {
+			// also nothing to change
+			memoryNeedToChange = false
+		}
+		if d.Get("cpu_hot_add_enabled").(bool) && d.HasChange("cpus") {
+			cpusNeedToChange = false
+		} else if !d.HasChange("cpus") {
+			// also nothing to change
+			cpusNeedToChange = false
+		}
```

Comment:
The above if blocks are very difficult to read. Can this be simplified? If not, please add a comment. Question rise:
- Why if hot add enabled and memory/cpu **has** changed, the memory/cpusNeedToChange is set to **false** (one would expect to true).

Reply:
changed polarity of this case, now looks simpler